NeoML/src/Dnn/DnnSimpleAPI.cpp

```diff
@@ -0,0 +1,789 @@
+		result->SetOutputCounts( outputCounts );
+	} );
+}
+
+CLayerWrapper<CSplitChannelsLayer> SplitChannels( int output0, int output1, int output2 )
+{
+	return CLayerWrapper<CSplitChannelsLayer>( "SplitChannels", [=]( CSplitChannelsLayer* result ) {
+		if( output1 == 0 ) {
+			result->SetOutputCounts2( output0 );
+		} else if( output2 == 0 ) {
+			result->SetOutputCounts3( output0, output1 );
+		} else {
+			result->SetOutputCounts4( output0, output1, output2 );
+		}
+	} );
+}
+
+CLayerWrapper<CSplitDepthLayer> SplitDepth( const CArray<int>& outputCounts )
+{
+	return CLayerWrapper<CSplitDepthLayer>( "SplitDepth", [&outputCounts]( CSplitDepthLayer* result ) {
+		result->SetOutputCounts( outputCounts );
+	} );
+}
+
+CLayerWrapper<CSplitDepthLayer> SplitDepth( int output0, int output1, int output2 )
+{
+	return CLayerWrapper<CSplitDepthLayer>( "SplitDepth", [=]( CSplitDepthLayer* result ) {
+		if( output1 == 0 ) {
+			result->SetOutputCounts2( output0 );
+		} else if( output2 == 0 ) {
+			result->SetOutputCounts3( output0, output1 );
+		} else {
+			result->SetOutputCounts4( output0, output1, output2 );
+		}
+	} );
+}
+
+CLayerWrapper<CSplitHeightLayer> SplitHeight( const CArray<int>& outputCounts )
+{
+	return CLayerWrapper<CSplitHeightLayer>( "SplitHeight", [&outputCounts]( CSplitHeightLayer* result ) {
+		result->SetOutputCounts( outputCounts );
+	} );
+}
+
+CLayerWrapper<CSplitHeightLayer> SplitHeight( int output0, int output1, int output2 )
+{
+	return CLayerWrapper<CSplitHeightLayer>( "SplitHeight", [=]( CSplitHeightLayer* result ) {
+		if( output1 == 0 ) {
+			result->SetOutputCounts2( output0 );
+		} else if( output2 == 0 ) {
+			result->SetOutputCounts3( output0, output1 );
+		} else {
+			result->SetOutputCounts4( output0, output1, output2 );
+		}
+	} );
+}
+
+CLayerWrapper<CSplitWidthLayer> SplitWidth( const CArray<int>& outputCounts )
+{
+	return CLayerWrapper<CSplitWidthLayer>( "SplitWidth", [&outputCounts]( CSplitWidthLayer* result ) {
+		result->SetOutputCounts( outputCounts );
+	} );
+}
+
+CLayerWrapper<CSplitWidthLayer> SplitWidth( int output0, int output1, int output2 )
+{
+	return CLayerWrapper<CSplitWidthLayer>( "SplitWidth", [=]( CSplitWidthLayer* result ) {
+		if( output1 == 0 ) {
+			result->SetOutputCounts2( output0 );
+		} else if( output2 == 0 ) {
+			result->SetOutputCounts3( output0, output1 );
+		} else {
+			result->SetOutputCounts4( output0, output1, output2 );
+		}
+	} );
+}
+
+CLayerWrapper<C3dConvLayer> Conv3d( int filterCount,
+	const CConvAxisParams& heightParams, const CConvAxisParams& widthParams,
+	const CConvAxisParams& depthParams, bool isZeroFreeTerm )
+{
+	return CLayerWrapper<C3dConvLayer>( "Conv3d", [=]( C3dConvLayer* result ) {
+		result->SetFilterCount( filterCount );
+
+		result->SetFilterHeight( heightParams.Size );
+		result->SetPaddingHeight( heightParams.Padding );
+		result->SetStrideWidth( heightParams.Stride );
+		result->SetDilationHeight( heightParams.Dilation );

```

Comment:
I think it's better to remove `SetDilation*` calls from this function (mb it's good to check here that all dilations are equal to 1 and check/assert if not)